### PR TITLE
chore: convert examples to notebook

### PR DESCRIPTION
**Pull request by ThibaultFy**
# Description

For those who will have the courage to review this PRs... Thank you 🙏 

Changes:
- Convert examples to notebook
- Move examples to `docs/source/examples` to be visible when building the documentation
- Rm sphinx gallery dependency, in favour of nbsphinx. Nbsphinx needs accepts custom CSS files (in this PR). It also needs pandoc to be installed. The only way to install pandoc in RTD is with conda. We changed the behavior of RTD build in favour of a conda environment.
- Update examples requirements files
- Update example to pass from RST notation to markdown
- Binder is back 💪 

closes FL-1185

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: .binder/runtime.txt [hunk 1]
```diff
@@ -1 +1 @@
-python-3.8
+python-3.10
```
Comment (SdgJlbl) [inline .binder/runtime.txt:1]: 🎉

File: docs/environment.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: rtd
+channels:
+  - defaults
+  - conda-forge
+dependencies:
+  - python=3.8
```
Comment (SdgJlbl) [inline docs/environment.yml:6]: 🧐 
do we want python 3.10 here too?

**Review by SdgJlbl — APPROVED** [review]
Impressive PR, nicely done 😮 
Glad to be rid of `sphinx-gallery`, and good job on bringing Binder back from the dead. 
Reviewing the notebooks separately since I'm not sure how NBreview will interact with comments here.

File: docs/source/examples/substra_core/diabetes_example/run_diabetes.ipynb [hunk 3]
```diff
@@ -0,0 +1,725 @@
+{
```
Comment (SdgJlbl) [inline docs/source/examples/substra_core/diabetes_example/run_diabetes.ipynb:1]: <p>Nor this one, technically</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: docs/source/examples/substra_core/titanic_example/run_titanic.ipynb [hunk 4]
```diff
@@ -0,0 +1,568 @@
+{
```
Comment (SdgJlbl) [inline docs/source/examples/substra_core/titanic_example/run_titanic.ipynb:1]: <p>I think we are not using matplolib on this specific example</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

File: docs/source/examples/substrafl/get_started/run_mnist_torch.ipynb [hunk 5]
```diff
@@ -0,0 +1,797 @@
+{
```
Comment (SdgJlbl) [inline docs/source/examples/substrafl/get_started/run_mnist_torch.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    dependencies = Dependency(pypi_dependencies=["numpy==1.23.1", "torch==1.11.0", "scikit-learn==1.1.1"])
</pre><p>1.11? maybe we want to update to 1.13.1 (at least until we bump it to 2+)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

File: docs/source/examples/substrafl/go_further/run_diabetes_substrafl.ipynb [hunk 6]
```diff
@@ -0,0 +1,698 @@
+{
```
Comment (SdgJlbl) [inline docs/source/examples/substrafl/go_further/run_diabetes_substrafl.ipynb:1]: <p><code>Compute Plan Builder</code> and <code>Decorator</code> are not displaying correctly, for some reasons</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

File: docs/source/examples/substrafl/go_further/run_iris_sklearn.ipynb [hunk 7]
```diff
@@ -0,0 +1,679 @@
+{
```
Comment (SdgJlbl) [inline docs/source/examples/substrafl/go_further/run_iris_sklearn.ipynb:1]: <p>Ok, the missing coma is probably on all notebooks, given that this paragraph has been copy-pasted everywhere</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: docs/source/examples/substrafl/go_further/run_mnist_cyclic.ipynb [hunk 8]
```diff
@@ -0,0 +1,1113 @@
+{
```
Comment (SdgJlbl) [inline docs/source/examples/substrafl/go_further/run_mnist_cyclic.ipynb:1]: <p>If not, follow the instructions described <a href="https://docs.substra.org/en/stable/substrafl_doc/substrafl_overview.html#installation" target="_blank">here</a>. (missing coma)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (SdgJlbl) [inline docs/source/examples/substrafl/go_further/run_mnist_cyclic.ipynb:1]: <p>Link generation problem here too, for <code>TorchAlgorithm</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Substra/substra-documentation/pull/368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

File: docs/environment.yml (hunk 2 above)
Comment (ThibaultFy) [reply docs/environment.yml:6]: good catch, this is a leftover when I tried different config to make it work (the key was to select `mambaforge-22.9`  in the rtd.yaml

**Pull request merged**